File: csharp/grains/Grains.cs
using System;
using System.Linq;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n is <= 0 or > 64)
        {
            throw new ArgumentOutOfRangeException();
        }

        return 1UL << n - 1;
    }

    public static ulong Total() => Enumerable.Range(1, 64).Select(Square).Aggregate(0UL, (acc, n) => acc + n);
}

File: c/grains/grains.c
#include "grains.h"

uint64_t square(uint8_t index) {

    if (index == 0 || index > 64) {
        return 0;
    }

    return (uint64_t) 1 << (index - 1);


}


uint64_t total(void) {

    uint64_t tot = 0;
    for (uint8_t i = 1; i <= 64; i++) {
        tot += square(i);
    }

    return tot;

}
